### ISSUE TITLE
still login

### ISSUE THREAD

**Issue by alahammad** [moloch--/RootTheBox#373]
Hello,

After i login to system after i create user account, then i logout. but i press back i can see my account as i am login.

**Comment by jeffg2k** [comment]
I think that is just due to the browser cache.  If you refresh, it should send you back to the login screen.  Do you expect a different result?

**Comment by alahammad** [comment]
should be as you mentioned login screen, but i tried too browsers, and same issue

**Comment by jeffg2k** [comment]
Perhaps I'm not understanding, as I'm not able to recreate an issue.  If you could explain it in steps, what page you're viewing, and what you expect to happen.  The logout does a deauth and clears the cookie, so while the back button can pull from your local browser cache, any refresh on a page that requires authentication will get redirected to the logon page.

**Comment by alahammad** [comment]
I will tell the scenario, user create account successfully then he logout . after that he pressed back in browser. 
the normal behavior as you mentioned login screen again. but what happed in my case user back to registration screen like session didn't clear.

**Comment by jeffg2k** [comment]
The normal behavior is to see the page as it's in you local browser cache.  Only after a refresh, which queries the server, would you get a redirect.

So are you thinking it should prevent a user from clicking the back button and seeing the local browser's cache?  Perhaps JavaScript verification force redirect.  @moloch-- any thought on this?  I'm a little confused.

**Comment by alahammad** [comment]
the normal behavior as i see in other sites when user back after logout redirect him to login screen.

**Comment by jeffg2k** [comment]
I understand -  after doing some research, you're describing [this](https://stackoverflow.com/questions/21680881/tornado-web-authenticated-back-button-issue) and [this](https://www.codeproject.com/Tips/549347/Browser-Back-Button-Issue-After-Logout-2)
And many sites do employ various methods to prevent using the back button as they contain sensitive information.  There are quick options like disabling caching or breaking the back button, but I wouldn't want to do either of those.  The other alternative seems to be building in an ajax request so the cached page makes a query as part of the page load, sees that the server session is no longer valid and redirects.  This would have to be done on all authenticated pages.

Are you seeing this as an issue for a classroom environment (students trying to see hints of other students by clicking back) or is it just an observation?  While I appreciate the report, I'm not sure the work effort is worth it for this particular software.  I'm inclined to tag it as `wontfix`.

**Comment by jeffg2k** [comment]
Just to clarify (based on the issue title), you're not still logged in after logging out.  What you're seeing the local copy (cache) of your web browser as the back button reloads the page from your hard disk.  It doesn't query the server (or postback).  You wouldn't be able to do anything except see what's been cached on your local system.  Doing anything else would query the server and result in a redirect to login.

**Comment by alahammad** [comment]
Actually as you said it's issue the students use this to see other players information, i understand it's not show stopper issue but nice if we have quick fix for that.

**Comment by jeffg2k** [comment]
I was able to find a quick fix I think and pushed it.  There are probably ways to work around it using dev tools, zap or burps, but I think I found a simple approach which shouldn't cause any issues.  I tied it into the menu system, which has separate files for public, user, or admin - only loading the script on the user and admin menu.  It checks to see if the forward or back button was used to load the page and will force a refresh if true.  If the user is still logged in, it should still utilize most of the cache.  If not, the page will be redirected if logged out or load the appropriate authenticated page if they logged in with a different account.  Hope that helps and sorry for the initial confusion on the issue (hadn't really thought about how sites handle the back button).
